### PR TITLE
Add outline:none to .dialog

### LINKED ISSUE #10536: .dialog should have 'outline: none' on focus
**Describe the issue you encountered:** .dialog should have 'outline: none' on focus. Otherwise:

bug

- Platform (Win7, 8, 10? macOS? Linux distro?): Debian and Windows

- Brave Version (revision SHA): at lease since 0.17.19. the issue became noticeable as the dialog shadow was removed on 0.18.x. On 0.17.19 you get the orange focus ring as this shows:

  bug

  The issue has not been almost noticeable on Windows as the focus ring of the dialog appears exactly over the application window's border:
  
  bug

- Steps to reproduce:
    1. Open a site
    2. Click the lion icon
    
- Actual result: See above

- Expected result: the focus ring should be removed from `.dialog` on dialog.js

   https://github.com/brave/browser-laptop/blob/3d2b641cfb46170e75e6c20c40c6deb81713acd4/app/renderer/components/common/dialog.js#L74

- Will the steps above reproduce in a fresh profile? If not what other info can be added?

- Any related issues:

### PR DESCRIPTION
**Pull request by luixxiul**
Closes #10536

Auditors:

Test Plan:
1. Open brave.com
2. Click the lion icon
3. Make sure the focus ring around the browser's window does not appear

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by luixxiul** [comment]
The issue exists on macOS too, although it is almost unnoticeable. The color of the window border slightly changes by clicking the lion icon, as this gif image shows.

focusring

File: app/renderer/components/common/dialog.js [hunk 1]
```diff
@@ -44,8 +44,7 @@ class Dialog extends ImmutableComponent {
     return <div
       data-bookmark-hanger={this.props.bookmarkHanger}
       className={cx({
-        [css(styles.dialog)]: true,
-        [css(styles.dialog_isNotClickDismiss)]: !this.props.isClickDismiss,
+        [css(styles.dialog, !this.props.isClickDismiss && styles.dialog_isNotClickDismiss)]: true,
```
Comment (cezaraugusto) [inline app/renderer/components/common/dialog.js]: nit: let's just avoid this in the future, this is hard to read

Comment (luixxiul) [reply app/renderer/components/common/dialog.js]: I did so to ensure that the background for `dialog_isNotClickDismiss` on https://github.com/brave/browser-laptop/pull/10557/files#diff-5d3d90fdb0c3efa6acd09057f2d4b101R90 does overwrite the one for `dialog`, which does not exist yet.

Before this change, if you set background-color to `dialog`, sometimes the background-color for `dialog_isNotClickDismiss` would not be properly cascaded.

**Comment by luixxiul** [comment]
@bbondy should this not be added to 0.20.x ?

**Comment by luixxiul** [comment]
https://github.com/brave/browser-laptop/blob/0.20.x/app/renderer/components/common/dialog.js#L85

**Pull request merged**